exercises/diffie-hellman/diffie-hellman.spec.js

```diff
@@ -0,0 +1,67 @@
+import DiffieHellman from './diffie-hellman';
+
+describe('diffie-hellman', () => {
+  const p = 23;
+  const g = 5;
+  let diffieHellman = new DiffieHellman(p, g);
+
+  const alicePrivateKey = 6;
+  const alicePublicKey = 8;
+
+  const bobPrivateKey = 15;
+  const bobPublicKey = 19;
+
+  test('throws an error if the constructor arguments are out of range', () => {
+
+    expect(() => {
+      new DiffieHellman(0, 9999)
+    }).toThrow();
+
+  });
+
+  test('throws an error if the constructor arguments are not prime', () => {
+
+    expect(() => {
+      new DiffieHellman(10, 13)
+    }).toThrow();
+
+  });
+
+  test('throws an error if private key >= p-1', () => {
```

Comment:
`README` says:

> [...] picks a private key, `a`, greater than 1 and less than p.

Let `a` be the given private key, I'd write the specs for these cases:

1. if `a < 0`, throw (negative integers)
2. if `a == 0`, throw
3. if `a == 1`, throw
4. if `a > 1 && a < p`, do not throw. Maybe, this test case is covered in next specs so that it doesn't need a test here.
5. if `a == p`, throw
6. if `a > p`, throw